OpenEXR/IlmImf/ImfCompositeDeepScanLine.cpp

```diff
@@ -386,16 +385,17 @@ composite_line(int y,
            {
 
                float value = output_pixel[ _Data->_bufferMap[channel_number] ]; // value to write
-
+               intptr_t base = reinterpret_cast<intptr_t>(it.slice().base);
 
                 // cast to half float if necessary
                if(it.slice().type==OPENEXR_IMF_INTERNAL_NAMESPACE::FLOAT)
                {
-                   * (float *)(it.slice().base + y*it.slice().yStride + x*it.slice().xStride) = value;
+
+                   * reinterpret_cast<float*>(base + y*it.slice().yStride + x*it.slice().xStride) = value;
```

Comment:
These statements sure would be easier to read split in two, with a pointer variable assignment followed by the *ptr = value:

    float* ptr = reinterpret_cast<float*>(base + y*it.slice().yStride + x*it.slice().xStride);
    *ptr = value;